src/main/java/br/com/harbitech/school/category/Category.java

```diff
@@ -7,39 +7,57 @@
-    private String codeUrl;
-    @Column(columnDefinition = "TEXT")
-    private String description;
-    @Column(columnDefinition = "TEXT")
-    private String studyGuide;
-    @Enumerated(EnumType.STRING)
-    @Column(columnDefinition = "ENUM")
-    private CategoryStatus status = CategoryStatus.INACTIVE;
-    @Min(-1)
-    private int orderVisualization;
-    @Size(max = 400, message = "Ops! O caminho do ícone não deve ter mais do que 400 caracteres")
-    private String iconPath;
-    @Size(max = 7, message = "Ops! Uma cor em hexa decimal não tem mais que 7 caracteres")
-    private String htmlHexColorCode;
-    @OneToMany(mappedBy = "category")
-    private List<Subcategory> subCategories = new ArrayList<>();
+        @Id
+        @GeneratedValue(strategy = GenerationType.IDENTITY)
+        private Long id;
+        @NotBlank(message = "Por favor insira o nome da categoria")
+        @Size(max = 70, message = "Ops! O nome de uma categoria não pode ter mais do que 70 caracteres")
+        private String name;
+        @NotBlank(message = "Por favor insira o código da categoria")
+        @Size(max = 70, message = "Ops! O código de uma categoria não pode ter mais do que 70 caracteres")
+        @Pattern(regexp = "[-a-z]+", message = "O código da url da categoria está incorreto (só aceita letras minúsculas e hífen)")
+        private String codeUrl;
+        @Column(columnDefinition = "TEXT")
+        private String description;
+        @Column(columnDefinition = "TEXT")
+        private String studyGuide;
+        @Enumerated(EnumType.STRING)
+        @Column(columnDefinition = "ENUM")
+        private CategoryStatus status = CategoryStatus.INACTIVE;
+        @Min(-1)
+        private int orderVisualization;
+        @Size(max = 400, message = "Ops! O caminho do ícone não deve ter mais do que 400 caracteres")
+        private String iconPath;
+        @Size(max = 7, message = "Ops! Uma cor em hexa decimal não tem mais que 7 caracteres")
+        private String htmlHexColorCode;
+        @OneToMany(mappedBy = "category")
+        private List<Subcategory> subCategories = new ArrayList<>();
 
     public Category(){}
 
+    public Category(String name, String codeUrl, String description, CategoryStatus status, int orderVisualization,
+                    String iconPath, String htmlHexColorCode) {
+        this.name = name;
+        this.codeUrl = codeUrl;
+        this.description = description;
+        this.status = status;
+        this.orderVisualization = orderVisualization;
+        this.iconPath = iconPath;
+        this.htmlHexColorCode = htmlHexColorCode;
+    }
+
     public Category(String name, String codeUrl) {
+        validateNonBlankText(name, "O nome da categoria não pode estar em branco.");
+        validateNonBlankText(codeUrl, "O código da URL da categoria não pode estar em branco.");
+        validateUrl(codeUrl, "O código da url da categoria está incorreto (só aceita letras minúsculas e hífen): " + codeUrl);
```

Comment:
Dá pra trocar isso pelo `Assert` do Spring, do pacote `org.springframework.util`:

```suggestion
        Assert.hasText(name, "O nome da categoria não pode estar em branco.");
        Assert.hasText(codeUrl, "O código da URL da categoria não pode estar em branco.");
        Assert.isTrue(CodeUrlValidator.isValid(codeUrl), "O código da url da categoria está incorreto (só aceita letras minúsculas e hífen): " + codeUrl);
```

Aí você cria apenas um CodeUrlValidator, que é diferentão.

Pra verificar se tá nulo, dá pra fazer:

```java
Assert.isNotNull();
```

Referências:
- https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/util/Assert.html
- https://www.baeldung.com/spring-assert

Reply:
Opa ótimas melhorias Aquiles, apenas a do codeUrl que não sei se vale....pois dessa forma eu teria que fazer mudanças em todos os testes...nesse caso não seria melhor manter a classe apenas com esse método até para o caso de ocorrer outras validações mais complicadas, oque você acha?